.editorconfig

```diff
@@ -1,5 +1,169 @@
+# https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
+# https://editorconfig.org
+
+# top-most EditorConfig file
 root = true
 
-[*.{cs,xaml}]
+# Default settings:
+# A newline ending every file
+# Use 4 spaces as indentation
+[*]
+insert_final_newline = false
+indent_style = space
+indent_size = 4
+
+[project.json]
+indent_size = 2
+
+# C# files
+[*.cs]
 indent_style = tab
-csharp_space_after_keywords_in_control_flow_statements = true
\ No newline at end of file
+
+# Modifier preferences
+dotnet_style_require_accessibility_modifiers = never:suggestion
+
+# New line preferences
+csharp_new_line_before_open_brace = all
+csharp_new_line_before_else = true
+csharp_new_line_before_catch = true
+csharp_new_line_before_finally = true
+csharp_new_line_before_members_in_object_initializers = true
+csharp_new_line_before_members_in_anonymous_types = true
+csharp_new_line_between_query_expression_clauses = true
+
+# Indentation preferences
+csharp_indent_block_contents = true
+csharp_indent_braces = false
+csharp_indent_case_contents = true
+csharp_indent_switch_labels = true
```

Comment:
that makes 2 level of indentation for a switch, where it's logically one

Reply:
This is the current style in the code:

```csharp
switch (test)
{
    case 1:
        break;
}

Reply:
If we change the rules, then it will be a totally different style, unused before.

Reply:
I prefer this style, used in some places:
```csharp
switch(foo) {
case 1:
    break;
}
```

as it only make a single level of indentation for a single conditional. Oh wait, it require the opening brace to be on the same line ? Now my world start to make some sense...